### DISCUSSION TITLE
Add admin-side re-authenticate capability for sensitive actions

### DISCUSSION THREAD

**Discussion by drbyte** [zencart/zencart#5805] [Ideas]
Task: add re-authenticate capability to admin for situations like:
- [ ] friendlier admin page-timeouts
- [ ] SQL Patches
- [ ] admin activity log purging
- [ ] payment-refunds via admin order screen
- [ ] changing any PCI settings
- [ ] editing payment module settings such as API keys or passwords so these can't be inappropriately changed/viewed
- [ ] changing own admin password or name/email credentials
- [ ] changing other admins' password or name/email credentials

**Comment by jtwebsitedesign** [comment]
I would also suggest: when changing any admin user credentials (password, email, etc)

**Comment by drbyte** [comment]
Do you mean when changing one's own credentials, or anyone's?

**Comment by jtwebsitedesign** [comment]
Yes, I think doing both would be very secure.
